### PR TITLE
Feature/desired fill price

### PR DESCRIPTION
**Pull request by davidvuong**
Implements `desiredPrice` changes defined in SIP-2004 (https://sips.synthetix.io/sips/sip-2004/)

### CODE REVIEW

File: test/contracts/PerpsV2Market.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ const Status = {
 	PriceTooVolatile: 12,
 };
 
-contract('PerpsV2Market PerpsV2MarketAtomic', accounts => {
+contract.only('PerpsV2Market PerpsV2MarketAtomic', accounts => {
```
Comment (leomassazza) [inline test/contracts/PerpsV2Market.js:42]: rogue .only

File: test/contracts/PerpsV2Market.js [hunk 2]
```diff
@@ -789,7 +814,7 @@ contract('PerpsV2Market PerpsV2MarketAtomic', accounts => {
 					await transferMarginAndModifyPosition({
 						market: perpsV2Market,
 						account: trader2,
-						fillPrice: price,
+						price: price,
```
Comment (leomassazza) [inline test/contracts/PerpsV2Market.js:817]: if variable name is same you can ommit the assignation i.e. 
`price,`
but super np.

**Review by leomassazza — APPROVED** [review]
LGTM. Just one np (no need to update now) and one rogue `.only` might be hiding other errors :(

File: test/contracts/PerpsV2Market.js (hunk 2 above)
Comment (kaleb-keny) [reply test/contracts/PerpsV2Market.js]: Hey i don't know if it's possible, but keeping it as fill price would be preferable. I mean i often get confused going through the code on whether it's fillPrice or oraclePrice. this helps a bit

Comment (davidvuong) [reply test/contracts/PerpsV2Market.js]: Yeah no prob I can revert the name here back to `fillPrice`.

**Pull request merged**